browser/src/layer/tile/CanvasTileLayer.js

```diff
@@ -5370,68 +5368,21 @@ L.CanvasTileLayer = L.Layer.extend({
 
 		// May have been changed by _ensureContext garbage collection
 		var canvas = tile.canvas;
+		var needsUnpremultiply = true;
 
 		if (isKeyframe)
 		{
 			if (this._debugDeltas)
 				window.app.console.log('Applying a raw RLE keyframe of length ' + allDeltas.length +
-						       ' hex: ' + hex2string(allDeltas, allDeltas.length));;
+						       ' hex: ' + hex2string(allDeltas, allDeltas.length));
 
-			// Byte bashing fun
 			var width = canvas.width;
 			var height = canvas.height;
 
-			var resultu32 = new Uint32Array(width * height);
-			var resultu8 = new Uint8ClampedArray(resultu32.buffer, resultu32.byteOffset, resultu32.byteLength);
-
-			for (var y = 0; y < height; ++y)
-			{
-				var rleSize = allDeltas[offset] + allDeltas[offset+1] * 256;
-				offset += 2;
-				if (this._debugDeltas)
-					window.app.console.log('rle size ' + rleSize);
-
-				var rleMask = offset;
-				var rleMaskSizeBytes = 256/8;
-
-				offset += rleMaskSizeBytes;
-
-				var uniquePixels;
-				if (rleSize > 0)
-				{
-					this._unpremult.unpremultiply(allDeltas, rleSize * 4, offset);
-					uniquePixels = this._unpremult._pixels;
-					if (this._debugDeltas)
-						window.app.console.log(
-							'Pixels hex: ' + hex2string(this._unpremult._bytes, rleSize * 4));;
-				}
-
-				// It would be rather nice to have real 64bit types [!]
-				var lastPix = 0;
-				var lastMask = 0;
-				var bitToCheck = 256;
-				var rleMaskOffset = rleMask;
-
-				var pixOffset = y * width;
-				var pixSrc = 0;
-
-				for (var x = 0; x < width; ++x)
-				{
-					if (bitToCheck > 128)
-					{
-						bitToCheck = 1;
-						lastMask = allDeltas[rleMaskOffset++];
-					}
-					if (!(lastMask & bitToCheck))
-						lastPix = uniquePixels[pixSrc++];
-					bitToCheck = bitToCheck << 1;
-					resultu32[pixOffset++] = lastPix;
-				}
-
-				offset += rleSize * 4;
```

Comment:
The github UI unfortunately doesn't seem to allow comments on lines outside the diff, but I think this is where things go wrong, after this change the local "offset" var (declared line 5364) isn't incremented to point after the keyframe, so offset remains at 0 and the later while(offset < allDeltas.length) trundles it way through the keyframe data that it used to skip over.